Ghidra/Features/Decompiler/src/decompile/cpp/address.cc

```diff
@@ -646,9 +646,9 @@ void zero_extend(intb &val,int4 bit)
 
 {
   intb mask = 0;
-  mask = (~mask)<<bit;
```

Comment:
It's UB to left shift a negative value (which we get when we do `~0`). What we're trying to do is zero out everything after the bit at the index `bit` so we can achieve this in a different way.

Reply:
[23bd3d5](https://github.com/NationalSecurityAgency/ghidra/pull/3967/commits/23bd3d5672e81c6288f0585b7271a4f2366494eb)